### PR TITLE
feat(assetimage): adds getters for dimensions

### LINKED ISSUE #45: AssetFile should have methods for returning width/height of asset
Available from the 'details' array already. Having access to this will allow us to calculate the ratio of images which means we can use a placeholder technique to prevent page jank.

### PR DESCRIPTION
**Pull request by calumbrodie**
fixes #45

### CODE REVIEW

**Comment by shieldo** [comment]
These getters would need to be added to the interface, not just this particular concrete (async requests will provide an instance of `AssetPromise`, for example). They would also need to be added to other implementations of the interface. Due to the interface extension, it would be a breaking change.

**Comment by calumbrodie** [comment]
Are there implementations of the interface outwith this package? If so I'm not responsible for those?

Will implement on the interface within this package though.

**Comment by shieldo** [comment]
There are other implementations within this package. Any downstream implementations that existed would need to be updated once using any new version of the library. (I'm aware of some. Naturally, they wouldn't need to be changed prior to this change, but just noting that this is a breaking change.)

**Comment by calumbrodie** [comment]
@shieldo updated

File: src/Promise/AssetPromise.php [hunk 1]
```diff
@@ -103,4 +103,43 @@ protected function addRejectionHandlerToPromise(PromiseInterface $promise)
...
     }
+
+    /**
+     * @return int|null
+     */
+    public function getWidth()
+    {
+        $resolved = $this->getResolved();
+        if (!$resolved instanceof AssetInterface) {
+            return 0;
```
[the remark this answers is not in the document]
Comment (calumbrodie) [reply src/Promise/AssetPromise.php]: done

**Pull request merged**